### PR TITLE
optimize: optimize the github actions

### PR DESCRIPTION
**Pull request by wangliang181230**
1. optimize: optimize the github actions.
   优化所有actions的脚本。
2. 修复单元测试 `DataSourceProxyTest.getResourceIdTest()` 偶尔失败的问题。
3. 修复 `test.yml` 在 `spring-boot.version=2.2.13.RELEASE` 时，由于 `spring-framework.version` 版本不匹配，导致单测失败的问题。（之前的PR修复安全漏洞，固定了 `spring-framework.version` 导致）

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -154,7 +154,7 @@
             <id>image</id>
             <properties>
                 <image.publish.skip>false</image.publish.skip>
-                <image.tags>latest</image.tags>
```
Comment (slievrly) [inline pom.xml:157]: ${project.version} for release version

Comment (wangliang181230) [reply pom.xml:157]: Only release version need to add `${project.version}` tag?

Comment (wangliang181230) [reply pom.xml]: done

File: .github/workflows/publishes.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        uses: actions/checkout@v2.4.0
+
+      # step 2
+      - name: "Setup Java JDK"
+        uses: actions/setup-java@v2.5.0
+        with:
+          distribution: 'zulu'
+          java-version: 8
+          server-id: ossrh
+          server-username: OSSRH_USERNAME
```
Comment (slievrly) [inline .github/workflows/publishes.yml:28]: why need `server-username`?

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -30,20 +31,21 @@ jobs:
...
       - name: "Test with Maven"
         # https://docs.github.com/cn/free-pro-team@latest/actions/reference/context-and-expression-syntax-for-github-actions#github-context
-        run: if [ "${{ matrix.java }}" == "8" ]; then
-               ./mvnw -T 4C clean test -e                     -Dspring-boot.version=${{ matrix.springboot }} -DskipTests=false -Dcheckstyle.skip=false -Dlicense.skip=false -B -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn;
-             else
-               ./mvnw -T 4C clean test -e -Pargs-for-un-java8 -Dspring-boot.version=${{ matrix.springboot }} -DskipTests=false -Dcheckstyle.skip=false -Dlicense.skip=false -B -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn;
-             fi
+        run: |
+          if [ "${{ matrix.java }}" == "8" ]; then
+            ./mvnw -T 4C clean test                     -Dspring-boot.version=${{ matrix.springboot }} -Dcheckstyle.skip=false -Dlicense.skip=false -e -B -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn;
```
Comment (slievrly) [inline .github/workflows/test.yml:48]: `IMAGE_NAME` env not found.

File: .github/workflows/publishes.yml (hunk 2 above)
Comment (wangliang181230) [reply .github/workflows/publishes.yml:28]: This `server-username` will set to the `settings.xml` for `maven-deploy-plugin`

File: .github/workflows/publishes.yml [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+      # step 4
+      - name: "Package and Publish-Image-To-DockerHub"
+        if: matrix.type == 'image'
+        env:
+          REGISTRY_USERNAME: ${{ secrets.REGISTRY_USERNAME }}
+          REGISTRY_PASSWORD: ${{ secrets.REGISTRY_PASSWORD }}
+        run: |
+          if [ "${{github.ref}}" == "refs/heads/snapshot" ]; then
+            ./mvnw -T 4C clean package -Pimage               -DskipTests -e -B -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn;
```
Comment (slievrly) [inline .github/workflows/publishes.yml:54]: IMAGE_NAME env not found.

Comment (wangliang181230) [reply .github/workflows/publishes.yml:54]: Oh, I missed it.

File: .github/workflows/build.yml [hunk 5]
```diff
@@ -8,46 +8,44 @@ on:
...
-        # https://docs.github.com/cn/free-pro-team@latest/actions/reference/context-and-expression-syntax-for-github-actions#github-context
-        run: if [ "${{matrix.java}}" == "8" ] && [ "${{github.event_name}}" == "push" ] && [ "${{github.ref}}" == "refs/heads/develop" ]; then
-               ./mvnw -T 4C clean test -e -Pimage -DskipTests=false -Dcheckstyle.skip=false -Dlicense.skip=false -B -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn;
-             else
-               ./mvnw -T 4C clean test -e         -DskipTests=false -Dcheckstyle.skip=false -Dlicense.skip=false -B -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn;
-             fi
+        run: |
+          if [ "${{ matrix.java }}" == "8" ]; then
+            ./mvnw -T 4C clean test -Dcheckstyle.skip=false -Dlicense.skip=false -e -B -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn;
+          elif [ "${{ matrix.java }}" == "11" ]; then
```
Comment (slievrly) [inline .github/workflows/build.yml:44]: No essential differences are found between JDK8 and JDK11.

File: .travis.yml [hunk 6]
```diff
@@ -20,10 +20,11 @@ before_script:
     fi
 
 script:
-  - if [ "$TRAVIS_BRANCH" == "develop" ] && [ "$TRAVIS_PULL_REQUEST" == false ]; then
-      travis_wait 30 ./mvnw clean test -DskipTests=false -Dcheckstyle.skip=false -Dlicense.skip=false -P image -B -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn;
-    else
-      travis_wait 30 ./mvnw clean test -DskipTests=false -Dcheckstyle.skip=false -Dlicense.skip=false -B -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn;
+  - if [ "$TRAVIS_JDK_VERSION" == "openjdk8" ]; then
```
Comment (slievrly) [inline .travis.yml:23]: add TRAVIS_PULL_REQUEST

Comment (wangliang181230) [reply .travis.yml]: reverted

File: .github/workflows/build.yml (hunk 5 above)
Comment (wangliang181230) [reply .github/workflows/build.yml:34]: JDK11：`checkstyle.skip` and `license.skip` is `true`.

**Pull request merged**